### PR TITLE
feat: Added Radiogroup

### PR DESCRIPTION
**Pull request by CalumH0**
#2647425423

### CODE REVIEW

File: src/components/RadioGroup/RadioGroup.vue [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+} from '../../composables/forms.js';
+import { useEventHandler } from '../../composables/events.js';
+
+const formElement = useFormElementValue(props.label);
+
+const props = defineProps({
+  ...formElementProps,
+  modelValue: [String, Object, Array],
+  options: {
+    type: Array,
```
Comment (robiharid) [inline src/components/RadioGroup/RadioGroup.vue:50]: minor: simplify the prop input here
`options: { type: Array }`

suggestion: do you need default values if they are all falsy? won't they have if checks on usage?  or instead should you change them to `required` instead? 

for example i see props.formatted is not wrapped in v-if. perhaps the formatted should be required then

File: src/components/RadioGroup/RadioGroup.vue [hunk 2]
```diff
@@ -0,0 +1,100 @@
+<template>
+  <component
+    class="relative z-50"
+    :is="formElement ? CFormElement : CFragment"
+    v-bind="{ id, label, size, color, labelFormatter, message, stacked }"
+  >
+  <div :class="[ textSizeClass, disabledClass]">
+  <RadioGroup
```
Comment (robiharid) [inline src/components/RadioGroup/RadioGroup.vue:8]: minor: indentation here (RadioGroup is child of parent div)
suggestion: perhaps look at Prettier vscode plugin to auto format for you?

File: src/components/RadioGroup/RadioGroup.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
+<template>
+  <component
+    class="relative z-50"
+    :is="formElement ? CFormElement : CFragment"
+    v-bind="{ id, label, size, color, labelFormatter, message, stacked }"
+  >
+  <div :class="[ textSizeClass, disabledClass]">
+  <RadioGroup 
+  class="concrete__radiogroup"
```
Comment (robiharid) [inline src/components/RadioGroup/RadioGroup.vue:9]: minor: is concrete_radiogroup doing anything? I see no styles are applied so maybe it is better to remove it?

File: src/components/RadioGroup/RadioGroup.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  :class="layoutClass"
+  v-model="value" 
+  ref="inputref"
+  :disabled='disabled'
+  >
+    <RadioGroupOption v-for="option in options" :key="option" :value="option" v-slot="{ checked }">
```
Comment (robiharid) [inline src/components/RadioGroup/RadioGroup.vue:15]: comment: good use of scoped slots here :)

**Review by robiharid — COMMENTED** [review]
Awesome, looks really good! 
Just left a few comments on minor usage things, otherwise, great stuff 👍

File: src/components/RadioGroup/RadioGroup.vue (hunk 3 above)
Comment (danielstorey) [reply src/components/RadioGroup/RadioGroup.vue:9]: we've added semantic classes to all concrete component root nodes for debugging and testing purposes

File: src/components/RadioGroup/RadioGroup.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    isDirty.value = true;
+    localValue.value = value;
+    emit('update:modelValue', value);
+    onChange();
+  },
+});
+
+const isDirty = ref(false);
+const localValue = ref(null);
+const el = ref(null);
```
Comment (danielstorey) [inline src/components/RadioGroup/RadioGroup.vue:74]: This variable name needs to match the `ref` attribute in the template above (it's a magical vue3 thing).

I.e.  `const inputref = ref(null);` or camel case it as per Robi's suggestion

File: src/components/RadioGroup/RadioGroup.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+const emit = defineEmits(['update:modelValue', 'change']);
+const onChange = useEventHandler('change', props, emit, localValue, isDirty);
```
Comment (danielstorey) [inline src/components/RadioGroup/RadioGroup.vue:77]: Any inputs that use this composable should have the corresponding prop defined with a default value of `null`

**Review by danielstorey — CHANGES_REQUESTED** [review]

**Pull request merged**